hydra-cluster/src/Hydra/Generator.hs

```diff
@@ -23,65 +24,104 @@ networkId = Testnet $ NetworkMagic 42
 -- set.
 data Dataset = Dataset
   { fundingTransaction :: Tx
-  , transactionsSequence :: [Tx]
-  , signingKey :: SigningKey PaymentKey
+  , clientDatasets :: [ClientDataset]
   }
-  deriving (Show, Generic)
-
-defaultProtocolParameters :: ProtocolParameters
-defaultProtocolParameters = fromLedgerPParams ShelleyBasedEraShelley def
+  deriving (Show, Generic, ToJSON, FromJSON)
 
 instance Arbitrary Dataset where
-  arbitrary = sized (genConstantUTxODataset defaultProtocolParameters)
+  arbitrary = sized $ \n -> do
+    sk <- genSigningKey
+    genDatasetConstantUTxO sk defaultProtocolParameters (n `div` 10) n
```

Comment:
suggestion: explain the `n `div` 10` through a named constant or `where` definition. I don't understand what its meaning is

Reply:
question: why tie the number of clients to the sized parameter which represents the number of transactions? Those are different parameters of the benchmarks and right now we cannot support more clients than the number of nodes.

Reply:
AFAIK we only get a single `Int` from quickcheck using `sized` and this `Arbitrary` instance should generate some arbitrary dataset. So I figured for some standard `n = 30` or so we would get a dataset with 3 clients processing 30 transactions each.

Will put a comment with this explanation

Reply:
Yes, but we want the number of clients to be independent from the number of transactions because the former is somehow tied to the number of nodes in the cluster